src/test/java/ArraySortedBagEqualsTest.java

```diff
@@ -0,0 +1,60 @@
+import com.google.common.testing.EqualsTester;
+import org.junit.jupiter.api.Test;
+
+class ArraySortedBagEqualsTest {
+
+    @Test
+    @SuppressWarnings("UnstableApiUsage")
+    public void equals_verify_contract() {
+        ArraySortedBag<Integer> emptyA = new ArraySortedBag<>();
+        ArraySortedBag<Integer> emptyB = new ArraySortedBag<>();
+        ArraySortedBag<Integer> emptyC = new ArraySortedBag<>(1000);
+
+        ArraySortedBag<Integer> singletonA = new ArraySortedBag<>();
+        ArraySortedBag<Integer> singletonB = new ArraySortedBag<>();
+        ArraySortedBag<Integer> singletonC = new ArraySortedBag<>(1000);
+        singletonA.add(10);
+        singletonB.add(10);
+        singletonC.add(10);
+
+        ArraySortedBag<Integer> manyA = new ArraySortedBag<>();
+        ArraySortedBag<Integer> manyB = new ArraySortedBag<>();
+        ArraySortedBag<Integer> manyC = new ArraySortedBag<>(1000);
+        ArraySortedBag<Integer> manyDifferentOrder = new ArraySortedBag<>();
```

Comment:
Although the `ArraySortedBagTest` class deals with the ordering, there is no guarantee on how the data is actually stored in the container and how equals is defined. Or maybe i'm crazy, idk.

Reply:
Depends if the sorting is stable or not. If it's stable, you can assert equality on the elements in the bag (as `equals` does already). If it's unstable, `equals` should be implemented in terms of the bag's `Comparator`. Then, two bags which hold multiple equal (but not the same) objects will still be equal to each other, regardless of those objects' literal ordering.

Reply:
Realistically, `equals` should always be in terms of the `Comparator`, but /shrug

Reply:
So, what you're saying is, yeet `manyDifferentOrder`

Reply:
To make sure I understand what you mean:

1. If `T` implements `Comparable`, we know it's `compareTo` is good to go
2. It's possible `T` never overrided `equals`
3. Therefore, the `compareTo` is a more "defensive" way to check equality on `T` 

Thus, the `ArraySortedBag`'s equals could be written to be more "defeneive" by making use of the `compareTo`.

Reply:
Not exactly defencive, more consistency. When things are ordered by a comparator, it's reasonable to also compare equality based on that comparator. Asking questions like "is this thing in the collection?" may yield surprising results if the comparator is not consistent with `equals`.